.github/workflows/examples-nightly.yml

```diff
@@ -31,61 +31,28 @@ jobs:
-          sudo apt-get install -y build-essential
-          npm install
-          npm run build
-          npm install --prefix examples
-          npm install ts-node
+      - name: Build the docker containers
+        # Use a random string to avoid other providers on the same subnet which might cause tests to fail because it expects only providers named provider-1 and provider-2
+        run: docker compose -f tests/docker/docker-compose.yml build
 
-      - name: Configure python
-        continue-on-error: true
-        uses: actions/setup-python@v4
-        with:
-          python-version: "3.10"
+      - name: Start the docker containers
+        # Use a random string to avoid other providers on the same subnet which might cause tests to fail because it expects only providers named provider-1 and provider-2
+        run: docker compose -f tests/docker/docker-compose.yml down && docker compose -f tests/docker/docker-compose.yml up -d
 
-      - name: Install goth
-        run: |
-          pip install goth>=0.15.3
-          rm -rf ../goth/assets
-          python -m goth create-assets ../goth/assets
-          sed -Ezi 's/(use\-proxy:\s)(True)/\1False/mg' ../goth/assets/goth-config.yml
-          sed -Ezi 's/(use\-prerelease:\s)(false)/\1true\n    release-tag: "0.13.0-rc21"/mg' ../goth/assets/goth-config.yml
-          sed -i '/^ENTRYPOINT/i ENV YAGNA_AUTOCONF_APPKEY=try_golem' ../goth/assets/docker/yagna-goth-deb.Dockerfile
+      - name: Fund the requestor
+        # Use a funding script which will retry funding the requestor 3 times, else it exits with error. The faucet is not reliable and sometimes fails to fund the requestor, thus the retry.
+        run: sleep 4 && docker exec -t docker-requestor-1 /bin/sh -c "/golem-js/tests/docker/fund-requestor.sh"
 
-      - name: Cleanup Docker
-        if: always()
+      - name: Install and build the SDK in the docker container
         run: |
-          c=$(docker ps -q) && [[ $c ]] && docker kill $c
-          docker system prune -af
+          docker exec -t docker-requestor-1 /bin/sh -c "cd /golem-js && npm i && npm run build"
 
-      - name: Log in to GitHub Docker repository
-        run: echo ${{ secrets.GITHUB_TOKEN }} | docker login docker.pkg.github.com -u ${{github.actor}} --password-stdin
-
-      - name: Run test suite
-        env:
-          GITHUB_API_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      - name: Run the Examples tests
         run: |
-          npm run test:examples
-
-      - name: Upload test logs
-        uses: actions/upload-artifact@v2
-        if: always()
-        with:
-          name: goth-logs
-          path: /tmp/goth-tests
-
-      # Only relevant for self-hosted runners
-      - name: Remove test logs
-        if: always()
-        run: rm -rf /tmp/goth-tests
+          docker exec -t docker-requestor-1 /bin/sh -c "cd /golem-js && npm install --prefix examples && npm install ts-node && npm run test:examples -- --exitOnError"
```

Comment:
I think it's better to move the example installation (and ts-node) to the step above